internal/pkg/cli/job_delete.go

```diff
@@ -0,0 +1,206 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// SPDX-License-Identifier: Apache-2.0
+
+package cli
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	"github.com/aws/copilot-cli/internal/pkg/config"
+	"github.com/aws/copilot-cli/internal/pkg/term/color"
+	"github.com/aws/copilot-cli/internal/pkg/term/log"
+	"github.com/aws/copilot-cli/internal/pkg/term/prompt"
+)
+
+const (
+	jobDeleteNamePrompt              = "Which job would you like to delete?"
+	fmtJobDeleteConfirmPrompt        = "Are you sure you want to delete %s from application %s?"
+	fmtJobDeleteFromEnvConfirmPrompt = "Are you sure you want to delete %s from environment %s?"
+	jobDeleteConfirmHelp             = "This will remove the job from all environments and delete it from your app."
+	jobDeleteFromEnvConfirmHelp      = "This will remove the job from just the %s environment."
+)
+
+var (
+	errJobDeleteCancelled = errors.New("job delete cancelled - no changes made")
+)
+
+type deleteJobVars struct {
+	appName          string
+	skipConfirmation bool
+	name             string
+	envName          string
+}
+
+type deleteJobOpts struct {
+	deleteJobVars
+
+	// Interfaces to dependencies.
+	store  store
+	prompt prompter
+}
+
+func newDeleteJobOpts(vars deleteJobVars) (*deleteJobOpts, error) {
+	store, err := config.NewStore()
+	if err != nil {
+		return nil, fmt.Errorf("new config store: %w", err)
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &deleteJobOpts{
+		deleteJobVars: vars,
+
+		store:  store,
+		prompt: prompt.New(),
+	}, nil
+}
+
+// Validate returns an error if the user inputs are invalid.
+func (o *deleteJobOpts) Validate() error {
+	if o.appName == "" {
+		return errNoAppInWorkspace
+	}
```

Comment:
I think we can allow folks to delete their jobs outside of their workspace, since we don't delete their local manifest files.

What do you think of removing this statement and in `Ask` prompting for the app name if it's not set?

Reply:
If the app name isn't set in Validate, then GetJob won't be able to find the job for validation, though, right...?

Reply:
discussed offline and figured out a solution (to be applied to `svc_delete` as well)